File: src/main/java/priv/bajdcc/util/lexer/token/MetaType.java
package priv.bajdcc.util.lexer.token;

/**
 * 字符类型
 *
 * @author bajdcc
 */
public enum MetaType {
	CHARACTER('\0'), LPARAN('('), RPARAN(')'), STAR('*'), PLUS('+'), CARET('^'), QUERY(
			'?'), LSQUARE('['), RSQUARE(']'), BAR('|'), ESCAPE('\\'), DASH('-'), LBRACE(
			'{'), RBRACE('}'), COMMA(','), DOT('.'), NEW_LINE('\n'), CARRIAGE_RETURN(
			'\r'), BACKSPACE('\b'), DOUBLE_QUOTE('\"'), SINGLE_QUOTE('\''), END(
			'\0'), ERROR('\0'), NULL('\0'), MUST_SAVE('\0'), TERMINAL('`'), LT(
			'<'), GT('>'), SHARP('#');

	private char ch;

	MetaType(char ch) {
		this.ch = ch;
	}

	public char getChar() {
		return ch;
	}

	public void setChar(char ch) {
		this.ch = ch;
	}
}

File: src/main/kotlin/com/bajdcc/util/lexer/token/MetaType.kt
package com.bajdcc.util.lexer.token

/**
 * 字符类型
 *
 * @author bajdcc
 */
enum class MetaType constructor(var char: Char) {
    CHARACTER('\u0000'),
    LPARAN('('),
    RPARAN(')'),
    STAR('*'),
    PLUS('+'),
    CARET('^'),
    QUERY('?'),
    LSQUARE('['),
    RSQUARE(']'),
    BAR('|'),
    ESCAPE('\\'),
    DASH('-'),
    LBRACE('{'),
    RBRACE('}'),
    COMMA(','),
    DOT('.'),
    NEW_LINE('\n'),
    CARRIAGE_RETURN('\r'),
    BACKSPACE('\b'),
    DOUBLE_QUOTE('\"'),
    SINGLE_QUOTE('\''),
    END('\u0000'),
    ERROR('\u0000'),
    NULL('\u0000'),
    MUST_SAVE('\u0000'),
    TERMINAL('`'),
    LT('<'),
    GT('>'),
    SHARP('#')
}
